tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir

```diff
@@ -153,4 +153,11 @@
   // CHECK-NEXT: ebreak
   "riscv.ret"() : () -> ()
   // CHECK-NEXT: ret
+  "riscv.directive"() {"directive" = ".align", "value" = "2"} : () -> ()
+  // CHECK-NEXT: .align 2
+  "riscv.directive"() ({
+    %nested_addi = "riscv.addi"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+  }) {"directive" = ".text"} : () -> ()
+  // CHECK-NEXT:  .text
```

Comment:
I like the way we use Regions to indicate nesting.

Reply:
Agree using regions for directives looks really nice but I think it suggests some sort of *scoped* behavior and I'm not sure this is the case. For example:

```
"riscv.directive"() ({
    // block A
}) {"directive" = ".option norvc"} : () -> ()
// block B
```

If we consider the resulting asm being like:

```
.option norvc
    // block A
    // block B
```

The net result here is that instructions lowered from *both blocks* are affected by `norvc` (a.k.a. disable compressed ISA), while in the IR it looks like only stuff coming from the nested region would be `norvc`. It would be absolutely *amazing* to have proper *scoped* behavior but that would require automatic lowering of `.option push`/`.option pop` to manage the stack...

Reply:
@nazavode I agree, this is something to take into account. 
I think this also confused me when there were multiple consecutive directives, because in the IR they can be both nested, or not nested, and still affect each other...

That is a bit weird.

I think the main reason for using a region is for memory region directives?

How about we just use it for those cases where it makes sense, and keep an empty region, or a verifier that checks if its not a memory region directive?
i.e. we make a `ScopedDirectiveOp` class?
@superlopuh ?

Reply:
I have a feeling that this is mixing concepts a little bit. My understanding of the semantics of blocks in MLIR is that the guarantee is that if the first operation in a block is executed, they are all executed. There's also a notion of scope of ssa values. `.option norvc` is new to me, but it looks like a side effect, and there's no reason that something with side effects wouldn't affect things outside of its block. I might be missing something, but this seems out of scope (😉) for this PR. Something at a higher level should deal with this.